### PR TITLE
Adding the possibility to enter sim pin code for the modem gsm driver

### PR DESCRIPTION
**Pull request by youssefchei-xite**
@jukkar
After testing the gsm modem with different SIM cards, there is an issue with the SIM cards that contain PIN code.
This issue is fixed by this implementation.
this commit allows the modem the enter the pin code(if it is enabled) to access the SIM card functionalities.

### CODE REVIEW

**Review by marvinrr15 — APPROVED** [review]

File: drivers/modem/Kconfig.gsm [hunk 1]
```diff
@@ -61,4 +61,24 @@ config MODEM_GSM_MANUAL_MCCMNO
 	  the network provider and may need to be changed if auto is not
 	  selected.
 
+config MODEM_ENABLE_PIN
+	bool "Enable the pin code enter"
+	default n
```
Comment (jukkar) [inline drivers/modem/Kconfig.gsm:66]: The options are disabled by default so no need to have this line.

File: drivers/modem/Kconfig.gsm [hunk 2]
```diff
@@ -61,4 +61,24 @@ config MODEM_GSM_MANUAL_MCCMNO
...
+	help
+	  This will add the possibility to enter SIM pin code.
+
+	if MODEM_ENABLE_PIN
```
Comment (jukkar) [inline drivers/modem/Kconfig.gsm:70]: Instead of the `if`, it is better to have
```
config MODEM_SIM_PIN_CODE
     depends on MODEM_ENABLE_PIN
     ....
```

File: drivers/modem/gsm_ppp.c [hunk 3]
```diff
@@ -237,7 +262,7 @@ static const struct setup_cmd setup_cmds[] = {
 	SETUP_CMD_NOHANDLE("ATH"),
 	/* extender errors in numeric form */
 	SETUP_CMD_NOHANDLE("AT+CMEE=1"),
-
+
```
Comment (jukkar) [inline drivers/modem/gsm_ppp.c:265]: Unnecessary change in the line

**Review by jukkar — COMMENTED** [review]
The changes look reasonable, could you submit this PR to upstream?

**Comment by mbolivar-nordic** [comment]
Hello, please submit this PR upstream as requested by @jukkar
